Shared/Supporting Files/Extensions/Favorites.swift

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+extension String {
+    /// The key to read and write the names of the favorite samples to the user defaults.
+    static var favoriteSamplesKey: String {
+        "favoriteSamples"
+    }
```

Comment:
Does prefix `key` needed?

```suggestion
    static var favoriteSampleNames = "favoriteSampleNames"
```

Reply:
This there value to even having this if the title is the same as the value?

Reply:
This is just a key which will be stored in the UserDefaults and the value will be an array of favorite sample names. We are defining this in the extension because it is used at multiple places.